src/Spectre.Data/Datasets/BasicTextDataset.cs

```diff
@@ -0,0 +1,307 @@
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using Spectre.Data.Structures;
+
+namespace Spectre.Data.Datasets
+{
+    public class BasicTextDataset : IDataset
+    {
+
+        //TODO: Because of not-so failsafe constructors, make the Datasets 
+        //TODO: be eventually spitted out by some factory class!!!
+
+        #region Fields
+
+        /// <summary>
+        /// Metadata of the dataset.
+        /// </summary>
+        private Metadata _metadata;
+
+        /// <summary>
+        /// Container for storing spacial coordinates for every loaded spectrum.
+        /// </summary>
+        private List<SpatialCoordinates> _spatialCoordinates;
+
+        /// <summary>
+        /// Array of m/z values for all the spectras.
+        /// </summary>
+        private double[] _mz;
+
+        /// <summary>
+        /// Container for storing intensity values for every loaded spectrum.
+        /// </summary>
+        private List<double[]> _intensity;
+
+        #endregion
+
+        #region Constructors
+
+        /// <summary>
+        /// Constructor with file initialization.
+        /// </summary>
+        /// <param name="textFilePath">Path to the text file.</param>
+        public BasicTextDataset(string textFilePath)
+        {
+            CreateFromFile(textFilePath);
+        }
+        /// <summary>
+        /// Constructor with raw value initialization.
+        /// </summary>
+        /// <param name="mz">Array of m/z values.</param>
+        /// <param name="data">Array of intensity values.</param>
+        public BasicTextDataset(double[] mz, double[,] data)
+        {
+            CreateFromRawData(mz, data);
+        }
+
+        #endregion
+
+        #region IDataset
+
+        /// <summary>
+        /// See <see cref="IDataset"/> for description.
+        /// </summary>
+        public Metadata Metadata
+        {
+            get { return _metadata; }
+            private set { _metadata = value; }
+        }
+        /// <summary>
+        /// See <see cref="IDataset"/> for description.
+        /// </summary>
+        public IEnumerable<SpatialCoordinates> SpacialCoordinates
+        {
+            get { return _spatialCoordinates; }
+            private set { _spatialCoordinates = value as List<SpatialCoordinates>; }
+        }
+        /// <summary>
+        /// Method for creating new dataset from text file, overwriting current data.
+        /// </summary>
+        /// <param name="filePath">Path to a text file.</param>
+        /// <exception cref="Exception">Thrown where there is a problem with file loading.</exception>
+        public void CreateFromFile(string filePath)
+        {
+            try
+            {
+                StreamReader sr = new StreamReader(filePath);
```

Comment:
`StreamReader` is `IDisposable`. It should be enclosed into `using` clause, or at least cleaned up in `finally` block.
It is maintaned properly below, in `AppendFromFile`.